### PR TITLE
Survex 1.2.32 (new formula)

### PR DESCRIPTION
**Pull request by ArrEssJay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by RobDeBagel** [comment]
@ilovezfs the error for El Capitan is:
'/var/folders/kw/j7f63hdn1lj39702nh90xm9w0000gp/T/jenkins187866456339062361.sh: line 3: brew: command not found' 

Am I missing something or is this a problem with the test itself?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by RobDeBagel** [comment]
Hi, Is there something else I need to do here. Should I assume lack of action is a rejection of this pull request?

**Comment by ilovezfs** [comment]
we're just a bit busy.

File: Formula/survex.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  desc "Cave Surveying Tool"
+  homepage "https://www.survex.com"
+  url "https://survex.com/software/1.2.32/survex-1.2.32.tar.gz"
+  sha256 "ac15252618f6a59617ec1142609f04bba69b1f18b3490729e405ff89a4b22852"
+
+  depends_on "wxmac"
+  depends_on "proj"
+  depends_on "ffmpeg"
+
+  depends_on "gettext" => :build
```
Comment (fxcoudert) [inline Formula/survex.rb]: Build dep goes first, then others in alphabetical order

File: Formula/survex.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  def install
+    # Allow for non-static linking against PROJ EPSG/ESRI definitions
+    ["src/cavern.c", "src/aven.cc"].each { |f| inreplace f, /pj_set_finder\(msg_proj_finder\)/, "" }
```
Comment (fxcoudert) [inline Formula/survex.rb]: This needs to be fixed in the upstream project. We try to minimise the patching done in Homebrew.

File: Formula/survex.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    # Prevent assumption that this is a MacOS bundle build
+    inreplace "src/message.c", /#ifdef MACOSX_BUNDLE/, "#if 0"
```
Comment (fxcoudert) [inline Formula/survex.rb]: This also needs to be addressed upstream: this should be configurable by command-line option, not patching of source file.

File: Formula/survex.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                          "--bindir=#{bin}",
+                          "--mandir=#{man}",
+                          "--docdir=#{doc}",
+                          "--datadir=#{share}"
+
+    system "make"
+    system "make", "install"
+
+    # Create and populate Aven.app
+    mkdir_p ["Aven.app/Contents/MacOS", "Aven.app/Contents/Resources"]
```
Comment (fxcoudert) [inline Formula/survex.rb]: Same here: the package should be built from a makefile or a script, but we're not going to populate it ourselves in the formula. This is too fragile, likely to break at any point.

File: Formula/survex.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    File.open("Aven.app/Contents/PkgInfo", "w") { |file| file.write("APPLAVEN") }
+    cp "lib/Info.plist", "Aven.app/Contents"
+    # Aven has some hard-coded assumptions about relative binary locations
+    # Workaround until fixed upstream by symlinking these
+    ln_s ["#{bin}/aven", "#{bin}/cavern", "#{bin}/extend"], "Aven.app/Contents/MacOS"
+
+    mv "#{pkgshare}/images", "Aven.app/Contents/Resources"
+    mv "#{pkgshare}/unifont.pixelfont", "Aven.app/Contents/Resources/"
+
+    # Create MacOS resource icons
```
Comment (fxcoudert) [inline Formula/survex.rb]: Same here.

File: Formula/survex.rb [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    Dir["lib/icons/*.iconset.zip"].each do |isz|
+      system "unzip", "-d", "Aven.app/Contents/Resources", isz
+      isf = "Aven.app/Contents/Resources/"+isz[/([0-9a-zA-Z]*).iconset(?=.zip)/, 0]
+      system "iconutil", "--convert", "icns", isf
+      rm_rf isf
+    end
+
+    # Remove unnecessary shared resources
```
Comment (fxcoudert) [inline Formula/survex.rb]: If not needed, why are they created and installed?

File: Formula/survex.rb [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    EOS
+    pos = <<~EOS
+      ( Easting, Northing, Altitude )
+      (224177.87, 6254297.49,   812.00 ) test.0
+      (224177.87, 6254297.49,   802.00 ) test.1
+    EOS
+
+    system "#{bin}/cavern", (testpath/"test.svx")
+    system "#{bin}/3dtopos", (testpath/"test.3d")
+    File.open(testpath/"test.pos", "r") { |f| assert_equal f.read, pos }
```
Comment (fxcoudert) [inline Formula/survex.rb]: This is a very nice test! Thanks… not so many people write good tests

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Thanks for contributing the formula… it seems like the software is not really ready for inclusion, so I'd suggest you coordinate with upstream (via bug reports or other ways) to make it simpler to install, then we can include it.

**Comment by fxcoudert** [comment]
Actually, seeing that there is much to change upstream before we can update and accept this formula, I am going to close it for now. But we would very much welcome an update in the future when issues are fixed!

**Pull request closed without merging**